Mathlib/Tactic/Simproc/FinsetInterval.lean

```diff
@@ -0,0 +1,205 @@
+  avoids a quadratic time performance hit. -/
+simproc_decl Ioo_nat (Ioo _ _) := .ofQ fun u α e ↦ do
+  match u, α, e with
+  | 1, ~q(Finset ℕ), ~q(Ioo (OfNat.ofNat $em) (OfNat.ofNat $en)) =>
+    let some m := em.rawNatLit? | return .continue
+    let some n := en.rawNatLit? | return .continue
+    let ⟨es, p⟩ ← evalFinsetIccNat (m + 1) (n - 1) q($em + 1) q($en - 1)
+    return .done <| .mk es <| .some q(Ioo_eq_of_Icc_succ_pred_eq $p)
+  | _, _, _ => return .continue
+
+/-- Simproc to compute `Finset.Iic b` when `b : ℕ`.
+
+**Warnings**:
+* With the standard depth recursion limit, this simproc can compute intervals of size 250 at most.
+* Make sure to exclude `Finset.insert_eq_of_mem` from your simp call when using this simproc. This
+  avoids a quadratic time performance hit. -/
+simproc_decl Iic_nat (Iic _) := .ofQ fun u α e ↦ do
+  match u, α, e with
+  | 1, ~q(Finset ℕ), ~q(Iic (OfNat.ofNat $en)) =>
+    let some n := en.rawNatLit? | return .continue
+    let ⟨es, p⟩ ← evalFinsetIccNat 0 n q(0) en
+    return .done <| .mk es <| .some q(Iic_eq_of_Icc_zero_eq $p)
+  | _, _, _ => return .continue
+
+/-- Simproc to compute `Finset.Iio b` when `b : ℕ`.
+
+**Warnings**:
+* With the standard depth recursion limit, this simproc can compute intervals of size 250 at most.
+* Make sure to exclude `Finset.insert_eq_of_mem` from your simp call when using this simproc. This
+  avoids a quadratic time performance hit. -/
+simproc_decl Iio_nat (Iio _) := .ofQ fun u α e ↦ do
+  match u, α, e with
+  | 1, ~q(Finset ℕ), ~q(Iio (OfNat.ofNat $en)) =>
+    let some n := en.rawNatLit? | return .continue
+    match n with
+    | 0 =>
+      have : $en =Q 0 := ⟨⟩
+      return .done <| .mk q(∅) <| .some q(Iio_zero)
+    | n + 1 =>
+      have en' := mkNatLitQ n
+      have : $en =Q $en' + 1 := ⟨⟩
+      let ⟨es, p⟩ ← evalFinsetIccNat 0 n q(0) q($en')
+      return .done  <| .mk es <| some q(Iio_succ_eq_of_Icc_zero_eq $p)
+  | _, _, _ => return .continue
+
+attribute [nolint unusedHavesSuffices] Iio_nat Ico_nat
+
+example : Icc 1 0 = ∅ := by simp only [Icc_nat]
+example : Icc 1 1 = {1} := by simp only [Icc_nat]
+example : Icc 1 2 = {1, 2} := by simp only [Icc_nat]
+
+example : Ico 1 1 = ∅ := by simp only [Ico_nat]
+example : Ico 1 2 = {1} := by simp only [Ico_nat]
+example : Ico 1 3 = {1, 2} := by simp only [Ico_nat]
+
+example : Ioc 1 1 = ∅ := by simp only [Ioc_nat]
+example : Ioc 1 2 = {2} := by simp only [Ioc_nat]
+example : Ioc 1 3 = {2, 3} := by simp only [Ioc_nat]
+
+example : Ioo 1 2 = ∅ := by simp only [Ioo_nat]
+example : Ioo 1 3 = {2} := by simp only [Ioo_nat]
+example : Ioo 1 4 = {2, 3} := by simp only [Ioo_nat]
+
+example : Iic 0 = {0} := by simp only [Iic_nat]
+example : Iic 1 = {0, 1} := by simp only [Iic_nat]
+example : Iic 2 = {0, 1, 2} := by simp only [Iic_nat]
+
+example : Iio 0 = ∅ := by simp only [Iio_nat]
+example : Iio 1 = {0} := by simp only [Iio_nat]
+example : Iio 2 = {0, 1} := by simp only [Iio_nat]
```

Comment:
These belong in a (standalone) test file.

Reply:
I still don't understand the point of making standalone test files for things that already function the way they're intended to work by the end of their file:
1. Tests can already be made at the end of the file
2. One gets instantaneous feedback when modifying the thing/fixing breakage
3. The tests demonstrate to the wandering user how to use the thing
4. The tests don't weigh in the oleans + those specific ones are instantaneous

Reply:
Can you make a Zulip thread making these points?